### DISCUSSION TITLE
Application crashes when typing anything in ID/Relay Server fields

### DISCUSSION THREAD

**Discussion by C-Duv** [rustdesk/rustdesk#7766] [General]
### Bug Description

The v1.2.4 (65f7541ec76) application crashes the second I press any letter/digit key when any of the _Settings_ > _Network_ fields has focus (e.g. "_Server ID_").

### How to Reproduce

1. Install the v1.2.4 (65f7541ec76) macOS application
2. Start the application
3. Go to _Settings_
4. Go to _Network_
5. Unlock the pane using you administrative credentials
6. Click on one of the 4 input fields of the "ID/Relay Server" part:
   * ID Server
   * Relay Server
   * API Server
   * Key
7. Press any non-space key of your keyboard (as if you wanted to set an address or a key): a digit, a letter, a special character

Application crashes and macOS offers to send a report (window "RustDesk quit unexpectedly."

### Expected Behavior

Application should not crash and I should be able to type _ID/Relay Server_ settings

### Operating system(s) on local side and remote side

macOS 13.5 ("Ventura")

### RustDesk Version(s) on local side and remote side

1.2.4 (65f7541ec76)

### Screenshots

Here is the Apple crash report: [nightly_crash_report.txt](https://github.com/rustdesk/rustdesk/files/15027414/nightly_crash_report.txt)

### Additional Context

The fields in the Socks5 Agents have no issue.

**Comment by rustdesk** [comment]
Apologize, we can not reproduce it on osx 14. Sadly, we have no osx 13 to test.

**Reply by C-Duv** [reply]
I just updated to macOS v14.4.1 ("Sonoma"): same issue, here is the report:

[rustdesk_v1.2.4-crash_report.txt](https://github.com/rustdesk/rustdesk/files/15031431/rustdesk_v1.2.4-crash_report.txt)

**Reply by C-Duv** [reply]
@rustdesk since I provided a report for newer macOS (v14) could you move this issue back to the issues and remove the _unreproducible_ label?

**Reply by rustdesk** [reply]
Few things we can do since it is underlying flutter engine problem, it has to be pending because we can not reproduce and we have so many reproducable issues in the list, :(. Hope you can understand. I totally understand, it makes you crazy.

# Please use `copy/paste` or `clipboard import` as a workaround.

**Reply by rustdesk** [reply]
Two textfield focus bug fixes in 3.22, https://groups.google.com/g/flutter-announce/c/0PEE5AvDZqc, not sure if it is related.

**Reply by rogerswb** [reply]
Error is present on macOS 12.7.6, also. Copy/paste and clipboard import doesn't work, either.

```
2024-09-19 13:51:20.376 RustDesk[74043:3962270] *** Terminating app due to uncaught exception 'NSInvalidArgumentException', reason: '-[AXPlatformNodeCocoa startEditing]: unrecognized selector sent to instance 0x600000354cc0'
*** First throw call stack:
(
	0   CoreFoundation                      0x00007ff80471c6e3 __exceptionPreprocess + 242
	1   libobjc.A.dylib                     0x00007ff80447c8bb objc_exception_throw + 48
	2   CoreFoundation                      0x00007ff8047a132b -[NSObject(NSObject) __retain_OA] + 0
	3   CoreFoundation                      0x00007ff804683467 ___forwarding___ + 1356
	4   CoreFoundation                      0x00007ff804682e88 _CF_forwarding_prep_0 + 120
	5   FlutterMacOS                        0x0000000102e47c46 FlutterMacOS + 7238
	6   FlutterMacOS                        0x0000000102e4759b FlutterMacOS + 5531
	7   FlutterMacOS                        0x000000010373c94d InternalFlutterGpu_Texture_AsImage + 31581
	8   FlutterMacOS                        0x0000000102e77c83 FlutterMacOS + 203907
	9   FlutterMacOS                        0x0000000102e5b73f FlutterMacOS + 87871
	10  FlutterMacOS                        0x000000010374f658 InternalFlutterGpu_Texture_AsImage + 108648
	11  FlutterMacOS                        0x0000000103762343 InternalFlutterGpu_Texture_AsImage + 185683
	12  FlutterMacOS                        0x00000001036838cd _ZdaPvSt11align_val_t + 8215165
	13  FlutterMacOS                        0x000000010375f891 InternalFlutterGpu_Texture_AsImage + 174753
	14  FlutterMacOS                        0x00000001037486a0 InternalFlutterGpu_Texture_AsImage + 80048
	15  FlutterMacOS                        0x0000000102e5b074 FlutterMacOS + 86132
	16  FlutterMacOS                        0x0000000102e5b220 FlutterMacOS + 86560
	17  libdispatch.dylib                   0x00007ff8044210cc _dispatch_call_block_and_release + 12
	18  libdispatch.dylib                   0x00007ff804422317 _dispatch_client_callout + 8
	19  libdispatch.dylib                   0x00007ff80442ec78 _dispatch_main_queue_drain + 943
	20  libdispatch.dylib                   0x00007ff80442e8bb _dispatch_main_queue_callback_4CF + 31
	21  CoreFoundation                      0x00007ff8046dee67 __CFRUNLOOP_IS_SERVICING_THE_MAIN_DISPATCH_QUEUE__ + 9
	22  CoreFoundation                      0x00007ff80469feff __CFRunLoopRun + 2771
	23  CoreFoundation                      0x00007ff80469ed6c CFRunLoopRunSpecific + 562
	24  HIToolbox                           0x00007ff80d3515e6 RunCurrentEventLoopInMode + 292
	25  HIToolbox                           0x00007ff80d35134a ReceiveNextEventCommon + 594
	26  HIToolbox                           0x00007ff80d3510e5 _BlockUntilNextEventMatchingListInModeWithFilter + 70
	27  AppKit                              0x00007ff8070ddaa9 _DPSNextEvent + 927
	28  AppKit                              0x00007ff8070dc166 -[NSApplication(NSEvent) _nextEventMatchingEventMask:untilDate:inMode:dequeue:] + 1394
	29  AppKit                              0x00007ff8070ce818 -[NSApplication run] + 586
	30  AppKit                              0x00007ff8070a279a NSApplicationMain + 817
	31  RustDesk                            0x0000000101b726e9 main + 9
	32  dyld                                0x0000000102d9352e start + 462
)
libc++abi: terminating with uncaught exception of type NSException
```

**Reply by rogerswb** [reply]
The previous stack trace happened when trying to type in the field. This one is from trying to use the import via copy/paste feature:
```
2024-09-19 13:54:52.487 RustDesk[74204:3964771] *** Terminating app due to uncaught exception 'NSInvalidArgumentException', reason: '-[AXPlatformNodeCocoa startEditing]: unrecognized selector sent to instance 0x600002cc04a0'
*** First throw call stack:
(
	0   CoreFoundation                      0x00007ff80471c6e3 __exceptionPreprocess + 242
	1   libobjc.A.dylib                     0x00007ff80447c8bb objc_exception_throw + 48
	2   CoreFoundation                      0x00007ff8047a132b -[NSObject(NSObject) __retain_OA] + 0
	3   CoreFoundation                      0x00007ff804683467 ___forwarding___ + 1356
	4   CoreFoundation                      0x00007ff804682e88 _CF_forwarding_prep_0 + 120
	5   FlutterMacOS                        0x0000000106db3c46 FlutterMacOS + 7238
	6   FlutterMacOS                        0x0000000106db359b FlutterMacOS + 5531
	7   FlutterMacOS                        0x00000001076a894d InternalFlutterGpu_Texture_AsImage + 31581
	8   FlutterMacOS                        0x0000000106de3c83 FlutterMacOS + 203907
	9   FlutterMacOS                        0x0000000106dc773f FlutterMacOS + 87871
	10  FlutterMacOS                        0x00000001076bb658 InternalFlutterGpu_Texture_AsImage + 108648
	11  FlutterMacOS                        0x00000001076ce343 InternalFlutterGpu_Texture_AsImage + 185683
	12  FlutterMacOS                        0x00000001075ef8cd _ZdaPvSt11align_val_t + 8215165
	13  FlutterMacOS                        0x00000001076cb891 InternalFlutterGpu_Texture_AsImage + 174753
	14  FlutterMacOS                        0x00000001076b46a0 InternalFlutterGpu_Texture_AsImage + 80048
	15  FlutterMacOS                        0x0000000106dc7074 FlutterMacOS + 86132
	16  FlutterMacOS                        0x0000000106dc7220 FlutterMacOS + 86560
	17  libdispatch.dylib                   0x00007ff8044210cc _dispatch_call_block_and_release + 12
	18  libdispatch.dylib                   0x00007ff804422317 _dispatch_client_callout + 8
	19  libdispatch.dylib                   0x00007ff80442ec78 _dispatch_main_queue_drain + 943
	20  libdispatch.dylib                   0x00007ff80442e8bb _dispatch_main_queue_callback_4CF + 31
	21  CoreFoundation                      0x00007ff8046dee67 __CFRUNLOOP_IS_SERVICING_THE_MAIN_DISPATCH_QUEUE__ + 9
	22  CoreFoundation                      0x00007ff80469feff __CFRunLoopRun + 2771
	23  CoreFoundation                      0x00007ff80469ed6c CFRunLoopRunSpecific + 562
	24  HIToolbox                           0x00007ff80d3515e6 RunCurrentEventLoopInMode + 292
	25  HIToolbox                           0x00007ff80d35134a ReceiveNextEventCommon + 594
	26  HIToolbox                           0x00007ff80d3510e5 _BlockUntilNextEventMatchingListInModeWithFilter + 70
	27  AppKit                              0x00007ff8070ddaa9 _DPSNextEvent + 927
	28  AppKit                              0x00007ff8070dc166 -[NSApplication(NSEvent) _nextEventMatchingEventMask:untilDate:inMode:dequeue:] + 1394
	29  AppKit                              0x00007ff8070ce818 -[NSApplication run] + 586
	30  AppKit                              0x00007ff8070a279a NSApplicationMain + 817
	31  RustDesk                            0x0000000105f9d6e9 main + 9
	32  dyld                                0x000000010e3df52e start + 462
)
libc++abi: terminating with uncaught exception of type NSException
Abort trap: 6
```

**Reply by rogerswb** [reply]
It might also be worth mentioning that, if you first exit the RustDesk client, the settings in ~/Library/Preferences/com.carriez.RustDesk/RustDesk2.toml can be edited, and it works as expected unless the network settings are unlocked. At that point, the dots used to mask the actual connection information would normally be changed to reflect the real values, but for macOS, this causes the client to crash again.

**Reply by hywan9** [reply]
> Few things we can do since it is underlying flutter engine problem, it has to be pending because we can not reproduce and we have so many reproducable issues in the list, :(. Hope you can understand. I totally understand, it makes you crazy.
> 
> # Please use `copy/paste` or `clipboard import` as a workaround.

When I paste using cmd + v, it still crashes. If I use the clipboard, I won't enter anything.

**Reply by hywan9** [reply]
> It might also be worth mentioning that, if you first exit the RustDesk client, the settings in ~/Library/Preferences/com.carriez.RustDesk/RustDesk2.toml can be edited, and it works as expected unless the network settings are unlocked. At that point, the dots used to mask the actual connection information would normally be changed to reflect the real values, but for macOS, this causes the client to crash again.

Have you solved this problem now? Can I set the ID/Relay server by modifying the files in the ~/Library/Preferences/com.carriez.RustDesk? : )

**Reply by Rcpentuy** [reply]
@ke1ewang  haven't test whether it will take effect with direct modification but after configuration my file become the below format, perhaps you can have a try:
rendezvous_server = 'your_server_ip:the_port_which_remains_the_same'
nat_type = 1
serial = 0

[options]
relay-server = 'your_server_ip'
custom-rendezvous-server = 'your_server_ip'
key = 'you_key'
local-ip-addr = '192.168.x.x'

**Comment by Rcpentuy** [comment]
same issue on Mac sequoia 15.1

**Reply by Rcpentuy** [reply]
I managed to avoid this issue by downgrading to version 1.2.3

**Reply by hywan9** [reply]
I have a temporary solution here, you can install the previous version, such as [1.2.7](https://github.com/rustdesk/rustdesk/releases/tag/1.2.7), which will work normally, and then overwrite the installation with the latest version after configuring the settings.

**Reply by Rcpentuy** [reply]
thanks, i didn't know i can overwrite it with the new version after the configuration

**Reply by Accademia** [reply]
> I managed to avoid this issue by downgrading to version 1.2.3

https://github.com/rustdesk/rustdesk/releases/tag/1.2.3-2

“Indeed, version 1.2.3 does not crash. However, version 1.2.7 still crashes and is unusable.”

**Comment by rustdesk** [comment]
Another workaround https://github.com/rustdesk/rustdesk/discussions/10061

**Comment by rustdesk** [comment]
https://github.com/flutter/flutter/issues/151428 fixed in main of Flutter, but not published yet.

**Comment by rustdesk** [comment]
Could anyone try out **https://github.com/rustdesk/rustdesk/releases/nightly**?

**Reply by hywan9** [reply]
> **https://github.com/rustdesk/rustdesk/releases/nightly**

Still crashing.
<img width="1112" alt="image" src="https://github.com/user-attachments/assets/a3c6e00f-df8f-4272-bdec-789c18d699e2">

**Reply by rustdesk** [reply]
@21pages

**Reply by 21pages** [reply]
In addition to the TextFiled configuring the server config, are there any other TextFileds that will also cause a crash?

**Reply by rustdesk** [reply]
> In addition to the TextFiled configuring the server config, are there any other TextFileds that will also cause a crash?

@ke1ewang

**Reply by Accademia** [reply]
> Could anyone try out **https://github.com/rustdesk/rustdesk/releases/nightly**?

My situation here is still crashing as well. Thanks.

**Reply by hywan9** [reply]
> In addition to the TextFiled configuring the server config, are there any other TextFileds that will also cause a crash?

All TextFiled of ID/Relay server crash.
![CleanShot 2024-12-03 at 13 42 20](https://github.com/user-attachments/assets/13b2b408-432d-4425-a721-f2b420a89780)

**Reply by 21pages** [reply]
How about other TextFields in RustDesk? eg: Socks5/Http(s) proxy

**Reply by hywan9** [reply]
> How about other TextFields in RustDesk? eg: Socks5/Http(s) proxy

All of Socks5/Http(s) proxy are normal.

**Reply by rustdesk** [reply]
Could you try out the new https://github.com/rustdesk/rustdesk/releases/nightly?

**Reply by rustdesk** [reply]
> Could you try out the new https://github.com/rustdesk/rustdesk/releases/nightly?

@ke1ewang @Accademia

**Reply by hywan9** [reply]
> Could you try out the new https://github.com/rustdesk/rustdesk/releases/nightly?

Still crashes

**Reply by rustdesk** [reply]
Really, could you record a video?

**Reply by hywan9** [reply]
> Really, could you record a video?

Sorry, I noticed that I just installed the x86 version instead of the arm version, and now it works fine!

**Reply by Accademia** [reply]
> > Could you try out the new https://github.com/rustdesk/rustdesk/releases/nightly?
> 
> @ke1ewang @Accademia

Hello, I tried installing the latest version released six hours ago (2024-12-03 06:19) on Arm MacOS 15.1, and indeed, the crash issue no longer occurs.

I feel like the “Unlock Network Settings” button might have been the root cause of the problem. Previously, when the software crashed, I encountered two scenarios: one where entering characters after unlocking would cause the software to crash (the issue discussed in this post), and another where the software would crash immediately after unlocking and entering the password (without even seeing the interface for entering characters after unlocking). However, in this latest version, it seems like the “Unlock” button has been removed, which might have eliminated the root cause of the crashes.

I will continue to monitor, and if the crash happens again, I will update this thread. Thank you very much for your patience and contribution.

Additionally, I heard that the multi-ID/relay server feature was completed a few months ago, but it seems to be missing in the latest version. 

Thank you.
